neuralmonkey/attention/combination.py

```diff
@@ -349,19 +359,23 @@ def __init__(self,
         self.attentions = attentions
     # pylint: enable=too-many-arguments
 
+    @tensor
+    def batch_size(self) -> tf.Tensor:
+        return self.attentions[0].batch_size
+
     def initial_loop_state(self) -> HierarchicalLoopState:
         return HierarchicalLoopState(
             child_loop_states=[a.initial_loop_state()
                                for a in self.attentions],
-            loop_state=empty_attention_loop_state())
+            loop_state=empty_attention_loop_state(self.batch_size))
 
     # pylint: disable=too-many-locals
     def attention(self,
                   query: tf.Tensor,
                   decoder_prev_state: tf.Tensor,
                   decoder_input: tf.Tensor,
-                  loop_state: HierarchicalLoopState,
-                  step: tf.Tensor) -> Tuple[tf.Tensor, HierarchicalLoopState]:
+                  loop_state: HierarchicalLoopState) -> Tuple[tf.Tensor,
+                                                              HierarchicalLoopState]:
```

Comment:
od-entrovat před `self` a kouknout se, jestli se to nevejde. Jinak bych entroval za `Tuple[`